src/WorkerPool.cc

```diff
@@ -194,3 +194,35 @@ bool WorkerPool::WaitForResults(const Time &_timeout)
   }
   return signaled && !this->dataPtr->done;
 }
+
+//////////////////////////////////////////////////
+bool WorkerPool::WaitForResults(
```

Comment:
To avoid duplicating the logic, you could have the old function call this one by converting `Time` to `duration`.